src/node_url.cc

```diff
@@ -1748,6 +1755,9 @@ void URL::Parse(const char* input,
               return;
             }
             // the port is valid
+            if (IsDefaultSchemePort(url->scheme, static_cast<int>(port))) {
+              url->flags |= URL_FLAGS_IS_DEFAULT_SCHEME_PORT;
+            }
             url->port = NormalizePort(url->scheme, static_cast<int>(port));
```

Comment:
The job done by `IsDefaultSchemePort` is pretty similar to what `NormalizePort` does already. Can we instead check `url->port == -1` as an indicator for default scheme port?